loader/pgsql2shp-core.c

```
@@ -546,45 +546,79 @@ is_clockwise(int num_points, double *x, double *y, double *z)
 	}
 }
 
+static char *
+trim_user_query(char *usrquery)
+{
+	char *query = strdup(usrquery);
+	char *end = query + strlen(query);
```

Cppcheck (2.21.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)